src/components/Header/Header.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Header.scss';
+
+const { block, elem } = bem({
+    name: 'Header',
+    classnames: styles
+});
+
+const Header = props => {
+    const { logoSrc, logoLink, logoName, children, ...rest } = props;
+
+    return (
+        <div {...rest} {...block(props)}>
+            <div {...elem('wrapper', props)}>
+                <a href={logoLink}>
+                    <img src={logoSrc} alt={logoName} {...elem('logo', props)} />
```

Comment:
What if in **Search!** we have a component which is already a logo itself?

Assuming that logo should be something configurable (different customers – different logos etc), I don't see it as a part of `<Header />`.

You're putting too much responsibility to one component. That also relates to the question from @carlobernardini about onClick handling.

I'd like to be able to do this:
```js
import { SearchLogo } from 'some/common/search/components';

...

render() {
  return (
    <Header
      logo={<SearchLogo onClick={...} />}
    >
      some extra content  
    </Header>
  );
}





<IconSearch size={48} title="Search!" />

Reply:
What do you expect to happen if the logo supplied is larger then the header container? According to @carlobernardini Carlo it should have a fix height. If that is not a constrain, we can change the logo to be a single prop that accepts only node. Or do you want to keep the current options too?

Reply:
e.g. we can set a fixed height to the `<Header/>` and "tell" to the `<Logo/>` (via passing some prop for instance) that it should take all the vertical space.

Reply:
Ok